### PR TITLE
Reserve factor polygon v2

### PR DESCRIPTION
**Pull request by defijesus**
This AIP will update Reserve Factors (RF) on Polygon v2 to encourage users to migrate funds to v3.

### CODE REVIEW

File: src/AaveV2PolygonReserveFactorUpdate_20230717/UPDATE_RESERVE_FACTORS_POLYGON_V2.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+A list of relevant links like for this proposal:
+
+* [Governance Forum Discussion](https://governance.aave.com/t/arfc-reserve-factor-updates-polygon-aave-v2/13937)
+
+* [Test Cases](https://github.com/defijesus/aave-proposals/blob/reserve-factor-polygon-v2/src/AaveV2PolygonReserveFactorUpdate_20230717/AaveV2PolygonReserveFactorUpdate_20230717.t.sol)
+
+* [Payload Implementation](https://github.com/defijesus/aave-proposals/blob/reserve-factor-polygon-v2/src/AaveV2PolygonReserveFactorUpdate_20230717/AaveV2PolygonReserveFactorUpdate_20230717.sol)
+
+[Pre-Post Payload Protocol Diff](https://github.com/defijesus/aave-proposals/blob/reserve-factor-polygon-v2/diffs/preTestPolygonReserveFactorUpdate20230717_postTestPolygonReserveFactorUpdate20230717.md)
```
Comment (Zer0dot) [inline src/AaveV2PolygonReserveFactorUpdate_20230717/UPDATE_RESERVE_FACTORS_POLYGON_V2.md:51]:
```suggestion
* [Pre-Post Payload Protocol Diff](https://github.com/defijesus/aave-proposals/blob/reserve-factor-polygon-v2/diffs/preTestPolygonReserveFactorUpdate20230717_postTestPolygonReserveFactorUpdate20230717.md)
```

File: src/AaveV2PolygonReserveFactorUpdate_20230717/AaveV2PolygonReserveFactorUpdate_20230717.sol [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * @dev Update Reserve Factors (RF) on Polygon v2 to encourage users to migrate funds to v3.
+ * @author defijesus.eth - TokenLogic
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0xbe8f3f1df222a53d8ca05f8be0ec56aa9f169e9881f98e6534e078d155854713
+ * - Discussion: https://governance.aave.com/t/arfc-reserve-factor-updates-polygon-aave-v2/13937
+ */
+contract AaveV2PolygonReserveFactorUpdate_20230717 is AaveV2PayloadPolygon {
+
+  function _postExecute() internal override {
+    ILendingPoolConfigurator(AaveV2Polygon.POOL_CONFIGURATOR).setReserveFactor(AaveV2PolygonAssets.DAI_UNDERLYING, 26_00);
```
Comment (Zer0dot) [inline src/AaveV2PolygonReserveFactorUpdate_20230717/AaveV2PolygonReserveFactorUpdate_20230717.sol:34]: `AaveV2Polygon.POOL_CONFIGURATOR` should already be of type `ILendingPoolConfigurator`, so I think the casting can be removed, but it's inconsequential.

File: src/AaveV2PolygonReserveFactorUpdate_20230717/AaveV2PolygonReserveFactorUpdate_20230717.t.sol [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        assetsChanged[i] == AaveV2PolygonAssets.BAL_UNDERLYING ||
+        assetsChanged[i] == AaveV2PolygonAssets.CRV_UNDERLYING ||
+        assetsChanged[i] == AaveV2PolygonAssets.GHST_UNDERLYING ||
+        assetsChanged[i] == AaveV2PolygonAssets.LINK_UNDERLYING
+      ) {
+        continue;
+      }
+      ReserveConfig memory cfg = _findReserveConfig(allConfigsAfter, assetsChanged[i]);
+      _deposit(cfg, AaveV2Polygon.POOL, address(42), 100);
+    }
```
Comment (rustboyar) [inline src/AaveV2PolygonReserveFactorUpdate_20230717/AaveV2PolygonReserveFactorUpdate_20230717.t.sol:58]: we recommend to add e2e test:

```suggestion
    e2eTest(AaveV2Polygon.POOL);
```

File: src/AaveV2PolygonReserveFactorUpdate_20230717/AaveV2PolygonReserveFactorUpdate_20230717.t.sol [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    assetsChanged[3] = AaveV2PolygonAssets.WBTC_UNDERLYING;
+    assetsChanged[4] = AaveV2PolygonAssets.WETH_UNDERLYING;
+    assetsChanged[5] = AaveV2PolygonAssets.WMATIC_UNDERLYING;
+    assetsChanged[6] = AaveV2PolygonAssets.BAL_UNDERLYING;
+    assetsChanged[7] = AaveV2PolygonAssets.CRV_UNDERLYING;
+    assetsChanged[8] = AaveV2PolygonAssets.GHST_UNDERLYING;
+    assetsChanged[9] = AaveV2PolygonAssets.LINK_UNDERLYING;
+
+    _noReservesConfigsChangesApartFrom(allConfigsBefore, allConfigsAfter, assetsChanged);
+
```
Comment (rustboyar) [inline src/AaveV2PolygonReserveFactorUpdate_20230717/AaveV2PolygonReserveFactorUpdate_20230717.t.sol:46]: we recommend to add e2e test:

```suggestion

    e2eTest(AaveV2Polygon.POOL);
    
```

**Review by rustboyar — APPROVED** [review]

**Pull request merged**